app/presenters/unauthorized_presenter.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+class UnauthorizedPresenter
+  def initialize(params)
+    @params = params
```

Comment:
instead of capturing the params, build your GenericWork here:
``` ruby

class UnauthorizedPresenter
  attr_reader :asset
  def initialize(id)
    @asset = GenericWork.find(id) if GenericWork.exists?(id)
  end
```